zamba/models/cnnensemble_model.py

```diff
@@ -142,24 +152,38 @@ def _find_non_non_blank_frames(self):
                    fold=fold)
             single_frame_cnn.find_non_blank_frames(model_name=model_name, fold=fold)
 
-    def _find_blank_videos(self, file_names, dummy=False):
+    def _find_blank_videos(self, file_names, threshold, dummy=False):
         """Detects blank videos
 
         Args:
             file_names (list of str)
-            dummy (bool): If True, label all even-indexed videos as non-blank and all odd-indexed videos as blank
+            dummy (bool): If True, randomly assign blank probabilities. If False, assign a blank probability of 0 to
+                all videos
 
         Returns:
-            (set) A set of blank video file names
+            pd.Series: The probability that the video is blank for each video
         """
-        blank_videos = set()
+
+        def blank_nonblank_model(file_names):
+            rng = np.random.RandomState(100)
+            blank_probabilities = rng.rand(len(file_names))
+            return blank_probabilities
 
         if dummy:
-            for i, file_name in enumerate(file_names):
-                if (i % 2) == 1:
-                    blank_videos.add(file_name)
+            blank_probabilities = blank_nonblank_model(file_names)
+        else:
+            blank_probabilities = 0
+
+        blank_probabilities = pd.Series(
+            blank_probabilities,
+            index=[str(file_name) for file_name in file_names],
+            name="blank",
+        )
+
+        blank_file_names = blank_probabilities.loc[blank_probabilities > threshold].index
+        blank_file_names = set(Path(file_name) for file_name in blank_file_names)
 
-        return blank_videos
+        return blank_file_names, blank_probabilities
```

Comment:
If you return `blank_probabilities.to_frame()`, it should make the join etc. easier.